lib/mayaUsd/fileio/registryHelper.cpp

```diff
@@ -280,11 +282,28 @@ VtDictionary UsdMaya_RegistryHelper::GetComposedInfoDictionary(const std::vector
     return result;
 }
 
+std::vector<std::function<void()>> g_pythonUnloaders;
+
+static void PythonUnload(size_t unloaderIndex)
+{
+    g_pythonUnloaders[unloaderIndex]();
+    // No destruction to keep the order, there should not be a lot of elements
+}
+
+static bool g_pythonRegistry = false;
+
 /* static */
-void UsdMaya_RegistryHelper::AddUnloader(const std::function<void()>& func)
+void UsdMaya_RegistryHelper::AddUnloader(const std::function<void()>& func, bool fromPython)
 {
-    if (g_pythonRegistry)
+    if (fromPython) {
+        g_pythonUnloaders.emplace_back(func);
+        if (boost::python::import("atexit")
+                .attr("register")(&PythonUnload, g_pythonUnloaders.size() - 1)
```

Comment:
The magic is here, registering to atexit with static c++ function and g_pythonUnloaders current last element.